execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -271,19 +270,22 @@ where
                 for (_, pointer) in pointer_block.to_indexed_pointers() {
                     match pointer {
                         Pointer::LeafPointer(descendant_leaf_trie_key) => {
-                            trie_keys_to_visit_queue.push(descendant_leaf_trie_key)
+                            trie_keys_to_visit.push(descendant_leaf_trie_key)
                         }
                         Pointer::NodePointer(descendant_node_trie_key) => {
-                            trie_keys_to_visit_queue.push(descendant_node_trie_key)
+                            trie_keys_to_visit.push(descendant_node_trie_key)
                         }
                     }
                 }
             }
             // If we hit an extension block, add its pointer to the queue
             Some(Trie::Extension { pointer, .. }) => {
-                trie_keys_to_visit_queue.push(pointer.into_hash())
+                trie_keys_to_visit.push(pointer.into_hash())
             }
         }
+        // TODO Use BTreeSet when pop_first() is added to stable from nightly.
```

Comment:
Are we ok with leaving this TODO in here?

Reply:
Yes yes yes, because ideally we want to use a `HashSet` or `BTreeSet` and we are using `Vec`currently. The `pop_first()` stuff isn't stable yet, but it would be better if we used either of those two instead of `Vec`